files/stat.py

```diff
@@ -469,15 +519,30 @@ def main():
     # try to get mime data if requested
     if get_mime:
         output['mimetype'] = output['charset'] = 'unknown'
-        filecmd = [module.get_bin_path('file', True), '-i', path]
-        try:
-            rc, out, err = module.run_command(filecmd)
-            if rc == 0:
-                mimetype, charset = out.split(':')[1].split(';')
-                output['mimetype'] = mimetype.strip()
-                output['charset'] = charset.split('=')[1].strip()
-        except:
-            pass
+        mimecmd = [module.get_bin_path('file'), '-i', path]
+        if mimecmd:
```

Comment:
This conditional will always be true. The code should be something like:

```python
mimecmd = module.get_bin_path('file')
if mimecmd:
    mimecmd = [mimecmd, '-i', path]
```

If the command is not found, the module will fail.